davidklatch/wavepool-david-klatch
# Customizable news post ads

This branch adds the ability for users to create Companies and Advertisement in the Django CRM and link and advertisement to a specific News Post so that the ad appears on the post's page. In order to prevent deleting the default company and ad I overrode the delete_queryset admin function so that I could remove the default from the queryset, but I couldn't figure out how to get the success message to not include the default in the count.

To verify:
1. Run the command to load the company fixtures `python manage.py loaddata apps/company/fixtures/company.json`
2. Run the command to load the advertisement fixtures `python manage.py loaddata apps/advertising/fixtures/advertisement.json`
3. On the website go to a news post's page and verify that the Industry Dive ad is present. Then in the Django CRM navigate to the NewsPosts browse page and edit that news post to point to an Advertisement that is not the Industry Dive ad. Then on the website go back to that news post and verify that the ad is now the one you chose.
4. In the Django CRM navigate to the Companies browse page and create a Company
    * In the Django CRM navigate to the Advertisement browse page and create a Advertisement for that company
    * Point a news post to that ad and then verify that that ad appears on the news post's page on the web site
5. Point a NewsPost to an ad that is not the default Industry Dive ad.
    * Verify that that ad appears on the news post's page
    * Delete the ad
    * Verify that the news post's page now shows the default Industry Dive ad
6. Verify that the default Industry Dive ad and default company Industry Dive can't be deleted.
